worksheets.go

```diff
@@ -320,6 +333,33 @@ func (ws *Worksheet) Set(name string, value Value) error {
 		return fmt.Errorf("Set on slice field %s, use Append, or Del", name)
 	}
 
+	if field.constrainedBy != nil {
+		prevValue := ws.MustGet(name)
+
+		// plan rollback
+		hasFailed := true
+		defer func() {
+			if hasFailed {
+				ws.set(field, prevValue)
+			}
+		}()
+
+		err := ws.set(field, value)
+		if err != nil {
+			return err
+		}
+		constrainedByResult, err := field.constrainedBy.Compute(ws)
```

Comment:
Why do you constrain after setting the field?

Reply:
I didn't know how else to run the compute function and give it the new value.